cpp/src/arrow/dataset/file_base.h

```diff
@@ -130,6 +132,11 @@ class ARROW_DS_EXPORT FileFormat {
                           std::shared_ptr<ScanOptions> scan_options,
                           std::shared_ptr<ScanContext> scan_context,
                           std::unique_ptr<ScanTaskIterator>* out) const = 0;
+
+  /// \brief Open a fragment
```

Comment:
Hmm... can you explain why that's interesting in addition to `ScanFile`?

Reply:
It seems `ScanFile` could be implemented in terms of `MakeFragment` and `DataFragment::Scan`. Is there a reason this is not the case?

Reply:
I suspect you're right but we're still hammering out this API